test/gtest/tensor_2d_gen_ocl_hip.cpp

```diff
@@ -0,0 +1,370 @@
+            {{20, 16}, {16, 1}, {1, 1}, {1, 1}, {-1, 1, 1}},
+            {{20, 16}, {16, 1}, {20, 16}, {16, 1}, {1.0, 0.5, 0}},
+            {{20, 16}, {16, 1}, {1, 16}, {16, 1}, {1.0, 0.5, 0}},
+            {{20, 16}, {16, 1}, {20, 1}, {1, 1}, {1.0, 0.5, 0}},
+            {{20, 16}, {16, 1}, {1, 1}, {1, 1}, {1.0, 0.5, 0}},
+            {{32, 64}, {64, 1}, {32, 64}, {64, 1}, {1, 1, 0}},
+            {{32, 64}, {64, 1}, {1, 64}, {64, 1}, {1, 1, 0}},
+            {{32, 64}, {64, 1}, {32, 1}, {1, 1}, {1, 1, 0}},
+            {{32, 64}, {64, 1}, {1, 1}, {1, 1}, {1, 1, 0}},
+            {{32, 64}, {64, 1}, {32, 64}, {64, 1}, {-1, 1, 1}},
+            {{32, 64}, {64, 1}, {1, 64}, {64, 1}, {-1, 1, 1}},
+            {{32, 64}, {64, 1}, {32, 1}, {1, 1}, {-1, 1, 1}},
+            {{32, 64}, {64, 1}, {1, 1}, {1, 1}, {-1, 1, 1}},
+            {{32, 64}, {64, 1}, {32, 64}, {64, 1}, {1.0, 0.5, 0}},
+            {{32, 64}, {64, 1}, {1, 64}, {64, 1}, {1.0, 0.5, 0}},
+            {{32, 64}, {64, 1}, {32, 1}, {1, 1}, {1.0, 0.5, 0}},
+            {{32, 64}, {64, 1}, {1, 1}, {1, 1}, {1.0, 0.5, 0}}};
+}
+
+struct Op2DTensorGenericTest : public ::testing::TestWithParam<TensorsConfig>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle = get_handle();
+        tensorsConfig = GetParam();
+
+        tensA = tensor<float>{tensorsConfig.aclens, tensorsConfig.acstrides}.generate(
+            tensor_elem_gen_integer{MAX_TENSOR_ELEM});
+        tensB = tensor<float>{tensorsConfig.blens, tensorsConfig.bstrides}.generate(
+            tensor_elem_gen_integer{MAX_TENSOR_ELEM});
+        tensC = tensor<float>{tensorsConfig.aclens, tensorsConfig.acstrides}.generate(
+            tensor_elem_gen_integer{MAX_TENSOR_ELEM});
+
+        tensA_dev = handle.Write(tensA.data);
+        tensB_dev = handle.Write(tensB.data);
+
+        tensC_ocl = tensor<float>{tensorsConfig.aclens, tensorsConfig.acstrides};
+        tensC_hip = tensor<float>{tensorsConfig.aclens, tensorsConfig.acstrides};
+    }
+
+    void runOCL()
+    {
+        auto&& handle = get_handle();
+        tensC_dev     = handle.Write(tensC.data);
+        std::fill(tensC_ocl.begin(), tensC_ocl.end(), std::numeric_limits<float>::quiet_NaN());
+
+        std::string program_name = "MIOpenTensorKernels.cl";
+
+        auto first_not_one = std::find_if(
+            tensorsConfig.blens.rbegin(), tensorsConfig.blens.rend(), [](int i) { return i != 1; });
+        auto d = std::distance(tensorsConfig.blens.begin(), first_not_one.base());
+
+        int num_wg      = first_not_one != tensorsConfig.blens.rend()
+                              ? static_cast<int>(*first_not_one == 0 ? 1 : *first_not_one)
+                              : 1;
+        int work_per_wg = std::accumulate(tensorsConfig.aclens.begin() + d,
+                                          tensorsConfig.aclens.end(),
+                                          1,
+                                          std::multiplies<int>());
```

Comment:
I think the most part of this function is identical to runHIP. Could you merge all the preparations steps somewhere inside `SetUp` method or create any other methods for the common code.